### PR TITLE
Spinner

### PR DESCRIPTION
**Pull request by mridulnagpal**
#110 Spinner added

### CODE REVIEW

**Comment by jywarren** [comment]
Hmm, what seems to be the error here? Also, if you have a chance, do you think you could upload a screenshot? Thank you!!!

**Comment by jywarren** [comment]
Restarted the build -- seemed unrelated to your code!

**Comment by jywarren** [comment]
Hey this looks good! @mridulnagpal if you can upload a screenshot i'll merge it in. Thanks so much!

**Comment by mridulnagpal** [comment]
Hey @jywarren the spinner appears as soon as we check the button, but it needs to be removed when we are loaded with the location, is that right?

image

**Comment by mridulnagpal** [comment]
@jywarren Now it works fine!! Ready to merge, please have a look!!

File: examples/index.html [hunk 1]
```diff
@@ -25,8 +26,12 @@ <h1>Leaflet BlurredLocation <small>| by <a href="">Public Lab</a></small></h1>
       <hr />
 
       <h3>Enter a location</h3>
-      
-      <p><button class="btn btn-primary" onclick="blurredLocation.geocodeWithBrowser();">Get my location</button></p>
```
Comment (jywarren) [inline examples/index.html]: Ah, just take a look here at what a new contributor changed first -- this has become a button. Can you adapt your changes to a button? Thanks!!! This is going to be great.

Comment (mridulnagpal) [reply examples/index.html]: Not an issue just a moment!!

**Comment by mridulnagpal** [comment]
@jywarren Does this look fine?

image

File: src/blurredLocation.js [hunk 2]
```diff
@@ -281,10 +281,24 @@ BlurredLocation = function BlurredLocation(options) {
...
-      });
+    if(boolean) {
+      var label = document.createElement("label");
+      label.classList.add("spinner");
+      var i = document.createElement("i");
+      i.classList.add("fa");
+      i.classList.add("fa-spinner");
+      i.classList.add("fa-spin");
+      label.appendChild(i);
+      var element = document.getElementById("div1");
```
Comment (jywarren) [inline src/blurredLocation.js]: Hmm, should this be dynamically set, though? I think we perhaps should default to something like `ldi-geocode-button` -- what do you think?

Comment (mridulnagpal) [reply src/blurredLocation.js]: If we keep it static how do we render the spinner? Also changing the id

Comment (jywarren) [reply src/blurredLocation.js]: We can say:

```js
var geocodeButtonId = geocodeButtonId || "ldi-geocode-button";
var element = document.getElementById(geocodeButtonId);`
```

And add to the demo and documentation that you can either make a button with `id=ldi-geocode-button` OR override it by setting `geocodeButtonId` in the options. How's that? 

Thank you!!!!

**Comment by jywarren** [comment]
Let's look at this again too @mridulnagpal ! 👍

**Comment by mridulnagpal** [comment]
@jywarren Made the fixes and it passes as well, shall we merge it?

File: src/blurredLocation.js [hunk 3]
```diff
@@ -212,13 +214,27 @@ BlurredLocation = function BlurredLocation(options) {
...
     updateRectangleOnPan();
 
-  // function geocodeWithBrowser(boolean) {
-  //   if ("geolocation" in navigator) {
-  //     navigator.geolocation.getCurrentPosition(function(position) {
-  //     goTo(position.coords.latitude, position.coords.longitude,options.zoom);
-  //     });
-  //   }
-  // }
+  function geocodeWithBrowser(boolean) {
```
Comment (jywarren) [inline src/blurredLocation.js]: Let's call this Boolean "success"?

File: README.md [hunk 4]
```diff
@@ -98,6 +98,7 @@ We welcome contributions, and are especially interested in welcoming [first time
 |----------------|-----------------------------|---------|
 | latId          | the input to set latitude   | `'lat'` |
 | lngId          | the input to set longitude  | `'lng'` |
+| geocodeButtonId          | the division to wrapping "Get my location" for spinner  | `'ldi-geocode-button'` |
```
Comment (jywarren) [inline README.md]: "...for inserting spinner icons"

Comment (mridulnagpal) [reply README.md]: Done

File: src/blurredLocation.js (hunk 3 above)
Comment (mridulnagpal) [reply src/blurredLocation.js]: Done

**Comment by mridulnagpal** [comment]
@jywarren Done with the changes!!

**Comment by jywarren** [comment]
Hi, Mridul - can you rebase the changes to resolve the merge conflict, and post a screenshot? Great work! Eager to see this posted. 

I think I caused the merge conflict by adding a `width` meta tag so it renders more nicely on smartphones. Sorry!

**Comment by mridulnagpal** [comment]
@jywarren Is this correct?

**Comment by jywarren** [comment]
Would you mind doing a gif of this so I can see it? Thanks!

**Comment by mridulnagpal** [comment]
Here it is
spinner

**Comment by jywarren** [comment]
Super work here!!! Let's bump the version number by `v0.0.x` so this gets published to PublicLab.org next time!

**Pull request merged**